poly/src/polynomial/univariate/mod.rs

```diff
@@ -153,9 +153,20 @@ impl<'a, F: 'a + FftField> DenseOrSparsePolynomial<'a, F> {
                 Evaluations::from_vec_and_domain(evals, domain)
             },
             DPolynomial(Cow::Borrowed(d)) => {
-                Evaluations::from_vec_and_domain(domain.fft(&d.coeffs), domain)
+                let chunks = d.coeffs.chunks(domain.size());
+                let reduced = chunks.fold(vec![F::zero(); domain.size()], |x, y| {
+                    x.iter().zip(y).map(|(&x, y)| x + y).collect()
```

Comment:
```suggestion
                    cfg_iter!(x).zip(y).map(|(&x, y)| x + y).collect()
```

Reply:
Actually, can avoid extra allocations here by making this a for loop, like so:

```rust
let mut reduced = chunks[0].to_vec();
for chunk in chunks {
    cfg_iter_mut!(reduced).zip(chunk).for_each(|(x, y)| {
    *x += y;
});
}
```

Reply:
Thanks for the notes, should be better now.